r/R/dplyr-funcs-datetime.R

```diff
@@ -189,6 +189,65 @@ register_bindings_datetime <- function() {
   })
 }
 
+register_bindings_duration <- function() {
+  register_binding("difftime", function(time1,
+                                        time2,
+                                        tz,
+                                        units = c("auto", "secs", "mins",
+                                                  "hours", "days", "weeks")) {
+    units <- match.arg(units)
+    if (units != "secs") {
+      abort("`difftime()` with units other than seconds not supported in Arrow")
+    }
+
+    if (!missing(tz)) {
+      warn("`tz` is an optional argument to `difftime()` in R and will not be used in Arrow")
+    }
+
+    time1 <- build_expr("cast", time1, options = cast_options(to_type = timestamp()))
+    time2 <- build_expr("cast", time2, options = cast_options(to_type = timestamp()))
+
+    build_expr("cast", time1 - time2, options = cast_options(to_type = duration("s")))
+  })
+
+  register_binding("as.difftime", function(x,
+                                           format = "%X",
+                                           units = "auto") {
+    # windows doesn't seem to like "%X"
+    if (format == "%X" & tolower(Sys.info()[["sysname"]]) == "windows") {
+      format <- "%H:%M:%S"
+    }
+
+    if (units != "secs") {
+      abort("`as.difftime()` with units other than seconds not supported in Arrow")
+    }
```

Comment:
Same thing here, having the default always error is not great, let's make the default be `"secs"`